privacyidea/static/components/login/controllers/loginControllers.js

```diff
@@ -119,10 +119,15 @@ angular.module("privacyideaApp")
     });
 
     $scope.$on('IdleTimeout', function () {
-        console.log("Lock!");
-        $scope.logoutWarning = false;
-        $scope.$apply();
-        $scope.lock_screen();
+        if ($scope.timeout_action == "logout") {
+            console.log("Logout!");
+            $scope.logout();
+        } else if ($scope.timeout_action == "lockscreen") {
```

Comment:
I agree with that idea. There should be a default when the policy is not set. I will modify the Pull Request.